app/main/views/frameworks.py

```diff
@@ -33,8 +35,9 @@
 @flask_featureflags.is_active_feature('GCLOUD7_OPEN')
 def framework_dashboard(framework_slug):
     template_data = main.config['BASE_TEMPLATE_DATA']
-    # TODO add a test for 404 if framework doesn't exist
     framework = data_api_client.get_framework(framework_slug)['frameworks']
+    if not framework:
```

Comment:
Is this check necessary here?  If the framework doesn't exist then I don't think this line will ever be reached - I think either it will 404 from an HTTPError from the API client, or there will be a KeyError trying to dereference 'frameworks' if nothing was returned?

But I might be wrong.